File: store/src/main/java/com/nytimes/android/external/store/base/Fetcher.java
package com.nytimes.android.external.store.base;

import javax.annotation.Nonnull;

import rx.Observable;

/**
 * Interface for fetching new data for a Store
 *
 * @param <Raw> data type before parsing
 */
public interface Fetcher<Raw, Key> {

    /**
     * @param key Container with Key and Type used as a request param
     * @return Observable that emits {@link Raw} data
     */
    @Nonnull
    Observable<Raw> fetch(@Nonnull Key key);
}

File: store/src/main/java/com/nytimes/android/external/store3/base/Fetcher.kt
package com.nytimes.android.external.store3.base

import io.reactivex.Observable
import io.reactivex.Single


/**
 * Interface for fetching new data for a Store
 *
 * @param <Raw> data type before parsing
</Raw> */
interface Fetcher<Raw, Key> {

    /**
     * @param key Container with Key and Type used as a request param
     * @return Observable that emits [Raw] data
     */
    suspend fun fetch(key: Key):Raw
}
